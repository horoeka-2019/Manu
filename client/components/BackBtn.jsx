import React from 'react'
import { Button } from 'semantic-ui-react'

const BackBtn = () => <Button size='big' className='backBtnStyle'>Back</Button>

export default BackBtn
