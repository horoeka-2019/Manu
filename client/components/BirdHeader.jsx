import React from 'react'
import { Menu, Container } from 'semantic-ui-react'

class BirdHeader extends React.Component {
  render () {
    return (

      <Menu className='centerHeader' borderless fixed='top' inverted style = {{ backgroundColor: '#F3A712' }}>
        <Container>
          <Menu.Item>
            He Pokai Manu
          </Menu.Item>
        </Container>
      </Menu>

    )
  }
}

export default BirdHeader
