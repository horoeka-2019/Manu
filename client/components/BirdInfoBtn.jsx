import React from 'react'
import { Button } from 'semantic-ui-react'

const BirdInfoBtn = () => <Button size='huge' className='birdInfoBtnStyle'>Bird Info</Button>

export default BirdInfoBtn
