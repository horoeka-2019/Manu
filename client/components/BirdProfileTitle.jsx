import React from 'react'

class BirdProfileTitle extends React.Component {
  render () {
    return (
      <React.Fragment>

        <div className='birdProfileTitle'>Tui</div>

      </React.Fragment>
    )
  }
}

export default BirdProfileTitle
