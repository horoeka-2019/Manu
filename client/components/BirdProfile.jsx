import React from 'react'
import BirdHeader from './BirdHeader'
import FoundBtn from './FoundBtn'
import BirdInfoBtn from './BirdInfoBtn'
import BackBtn from './BackBtn'
import BirdProfileTitle from './BirdProfileTitle'
import { Link } from 'react-router-dom'
import { Segment, Grid } from 'semantic-ui-react'
import found from '../api/found'

class BirdProfile extends React.Component {
  render () {
    const { id } = this.props.match.params
    const bird = this.props.birds.find(bird => bird.bird_id === Number(id))
    return (
      <React.Fragment>

        <Segment style={{ padding: '5px 5px' }} vertical>
          <Grid container stackable verticalAlign='middle'>
            <Grid.Row>
              <BirdHeader />
              <div className='birdProfileImage'>{bird.image}</div>
              <BirdProfileTitle name={bird.name}/>
              <Link to="/">
                <FoundBtn onClick={found(id)}/>
              </Link>
              <Link to={`/profile/${id}/info`}>
                <BirdInfoBtn />
              </Link>
              <Link to="/">
                <BackBtn />
              </Link>
            </Grid.Row>
          </Grid>

        </Segment>

      </React.Fragment>
    )
  }
}

export default BirdProfile
