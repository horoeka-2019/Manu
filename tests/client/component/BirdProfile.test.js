import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import BirdProfile from '../../../client/components/BirdProfile'
import BirdHeader from '../../../client/components/BirdHeader'
import BirdProfileTitle from '../../../client/components/BirdProfileTitle'
import BackBtn from '../../../client/components/BackBtn'
import BirdInfoBtn from '../../../client/components/BirdInfoBtn'
import { Button } from 'semantic-ui-react'

const mockBirdProfileProps = {
  image: 'public/images/piwakawaka.jpg/'
}

describe('BirdProfile component test', () => {
  it('BirdProfile contains header, image, found, bird info and back button components', () => {
    // Arrange
    const image = mockBirdProfileProps
    const expected = [<BirdHeader/>, <img/>, image, <BirdProfileTitle />,
      <Button />,
      <BirdInfoBtn />,
      <BackBtn />]
    // Act
    const wrapper = mount(<BirdProfile/>)
    wrapper.setProps({ image: image })
    const actual = wrapper.containsAllMatchingElements(expected)
    // Assert
    expect(actual).toBeTruthy()
  })
  it('render unchanged from previous snapshot', () => {
    const wrapper = mount(<BirdProfile/>)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
