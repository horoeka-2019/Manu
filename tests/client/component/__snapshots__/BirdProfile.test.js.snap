// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Perching /> component test render unchanged from previous snapshot 1`] = `
<HashRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <BirdProfile
      match={
        Object {
          "params": Object {
            "id": "3",
          },
        }
      }
    />
  </Router>
</HashRouter>
`;
